enterprise/frontend/src/embedding-sdk/types/theme/index.ts

```diff
@@ -69,27 +69,39 @@ export type MetabaseColor = keyof MetabaseColors;
 /**
  * Theme options for customizing specific Metabase
  * components and visualizations.
+ *
+ * Every non-optional properties here must have a default value defined
+ * in DEFAULT_METABASE_COMPONENT_THEME at [default-component-theme.ts]
  */
 export interface MetabaseComponentTheme {
   /** Data tables **/
-  table?: {
-    cell?: {
-      /** Text color of cells, defaults to `text-dark`. */
-      textColor?: string;
+  table: {
+    cell: {
+      /** Text color of cells, defaults to `text-primary`. */
+      textColor: string;
```

Comment:
@deniskaber Good catch, I've just updated the PR so that all theme fields are **publicly optional** (the SDK user don't need to provide any theme options), but internally has defaults. I think I've accidentally unstaged the `DeepPartial<MetabaseTheme>` changes so that was missing from the PR 😅

```ts
export interface MetabaseTheme {
  /** Component theme options */
  components?: DeepPartial<MetabaseComponentTheme>;
}
```


The reason is that when we use `theme.other.table.cell.textColor` for instance, all types are optional even when they definitely has default values defined in `DEFAULT_METABASE_COMPONENT_THEME` in `default-component-theme.ts`, so we need to do `theme.other?.table?.cell?.textColor ?? <another default>` to extract the value.

I've discussed with @WiNloSt on [this comment thread](https://github.com/metabase/metabase/pull/42411#discussion_r1598625719) about having `DEFAULT_METABASE_COMPONENT_THEME` to be the single source of defaults, so by making some of them non-optional it's clear to us internally which one has explicitly-defined defaults.

```ts
export const DEFAULT_METABASE_COMPONENT_THEME: MetabaseComponentTheme = {
  table: {
    idColumn: {
      textColor: "brand",
    },
    cell: {
      textColor: "text-dark",
    },
  },
  pivotTable: {
    rowToggle: {
      textColor: "white",
      backgroundColor: "text-light",
    },
  },
};
```